setup.pl

```diff
@@ -991,3 +1010,19 @@ sub files_in_dir
 return @rv;
 }
 
+sub make_dir_recursive_local
+{
+my ($dir, $mod) = @_;
+my @folders = split /\//, $dir;
+map {
+    if ($_) {
+        if (mkdir $_) {
+            chmod($mod, $_) if ($mod && -d $_);
+            }
+        chdir $_;}
```

Comment:
I don't think calling chdir in here is a good idea, as that's a global operation that could effect the rest of the script.

Reply:
Then we could store the current `chdir` and restore it afterwards.

Reply:
But why chdir at all, rather than using absolute paths?

Reply:
To create a path dir-by-dir, without using `system` and `mkdir -p`.

Reply:
I mean why not call mkdir on the full path?

Reply:
We cannot simply call `mkdir` on the full path as it won't create parents, this is not a pleasant user experience.

Reply:
No I mean, instead of using `chdir` to enter each new directory, just always call `mkdir` on the full path.

Reply:
Oh, you mean, if the path is `/webmin/1.999-3/data/etc`, you want me to call it:

```
mkdir /webmin
mkdir /webmin/1.999-3
mkdir /webmin/1.999-3/data
mkdir /webmin/1.999-3/data/etc
```

.. without using `chdir` at all?

Reply:
Nice suggestion - fixed https://github.com/webmin/webmin/commit/82937d72963ff8309bd4914690dbf1f3b85aa168.

Reply:
Thanks! That works..